appservice/src/deploy/runWithZipStream.ts

```diff
@@ -33,42 +36,33 @@ export async function runWithZipStream(context: IActionContext, fsPath: string,
         });
     } else {
         ext.outputChannel.appendLog(localize('zipCreate', 'Creating zip package...'), { resourceName: client.fullName });
+        let filesToZip: string[] = [];
 
-        // level 9 indicates best compression at the cost of slower zipping. Since sending the zip over the internet is usually the bottleneck, we want best compression.
-        const zipper: archiver.Archiver = archiver('zip', { zlib: { level: 9 } });
         if ((await fse.lstat(fsPath)).isDirectory()) {
             if (!fsPath.endsWith(path.sep)) {
                 fsPath += path.sep;
             }
 
             if (client.isFunctionApp) {
-                await addFilesGitignore(zipper, fsPath, '.funcignore');
+                filesToZip = await getFilesFromGitignore(fsPath, '.funcignore');
             } else {
-                addFilesGlob(zipper, fsPath);
+                filesToZip = await getFilesFromGlob(fsPath);
             }
         } else {
-            zipper.file(fsPath, { name: path.basename(fsPath) });
+            filesToZip.push(path.basename(fsPath));
+            // remove the file from the fsPath
+            fsPath = path.dirname(fsPath);
+        }
+        const zipFile = new yazl.ZipFile();
+        for (const file of filesToZip) {
+            zipFile.addFile(path.join(fsPath, file), file);
         }
-        zipStream = zipper;
-        zipper.on('end', () => {
-            onFileSize(zipper.pointer());
```

Comment:
We lost this onFileSize output. If you can find a quick way to get it back that would be nice, but I would not spend much time on it